singularity-eos/eos/modifiers/ramps_eos.hpp

```diff
@@ -0,0 +1,239 @@
+  }
+  PORTABLE_FUNCTION
+  Real BulkModulusFromDensityInternalEnergy(const Real rho, const Real sie,
+                                            Real *lambda = nullptr) const {
+    const Real p_ramp{get_ramp_pressure(rho)};
+    const Real p_eos{t_.PressureFromDensityInternalEnergy(rho, sie, lambda)};
+
+    return p_eos < p_ramp ? rho * get_ramp_dpdrho(rho)
+                          : t_.BulkModulusFromDensityInternalEnergy(rho, sie, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real GruneisenParamFromDensityInternalEnergy(const Real rho, const Real sie,
+                                               Real *lambda = nullptr) const {
+    return t_.GruneisenParamFromDensityInternalEnergy(rho, sie, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real PressureFromDensityTemperature(const Real rho, const Real temperature,
+                                      Real *lambda = nullptr) const {
+    const Real p_ramp{get_ramp_pressure(rho)};
+    const Real p_eos{t_.PressureFromDensityTemperature(rho, temperature, lambda)};
+    return p_eos < p_ramp ? p_ramp : p_eos;
+  }
+  PORTABLE_FUNCTION
+  Real SpecificHeatFromDensityTemperature(const Real rho, const Real temperature,
+                                          Real *lambda = nullptr) const {
+    return t_.SpecificHeatFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real BulkModulusFromDensityTemperature(const Real rho, const Real temperature,
+                                         Real *lambda = nullptr) const {
+    const Real p_ramp{get_ramp_pressure(rho)};
+    const Real p_eos{t_.PressureFromDensityTemperature(rho, temperature, lambda)};
+    return p_eos < p_ramp
+               ? rho * get_ramp_dpdrho(rho)
+               : t_.BulkModulusFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real GruneisenParamFromDensityTemperature(const Real rho, const Real temperature,
+                                            Real *lambda = nullptr) const {
+    return t_.GruneisenParamFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  void FillEos(Real &rho, Real &temp, Real &energy, Real &press, Real &cv, Real &bmod,
+               const unsigned long output, Real *lambda = nullptr) const {
+    // output passed into internal filleos can't include pressure
+    const unsigned long ramp_out = output & ~thermalqs::pressure;
+    // if pressure is output, calculate it first
+    if (output & thermalqs::pressure) {
+      // maybe switch case on preferred input and check for not output of the input
+      // for now sie lookup is prioritized
+      if (!(output & thermalqs::specific_internal_energy)) {
+        press = this->PressureFromDensityInternalEnergy(rho, energy, lambda);
+      } else if (!(output & thermalqs::temperature)) {
+        press = this->PressureFromDensityTemperature(rho, temp, lambda);
+      }
+    }
+    // call internal filleos
+    t_.FillEos(rho, temp, energy, press, cv, bmod, ramp_out, lambda);
```

Comment:
Won't pressure be overwritten here?

Reply:
notice that I pass in the `ramp_out` which ensures pressure is not an output value.

Reply:
ah ok I see.